kernel/kernel-api/src/main/java/io/delta/kernel/client/ExpressionHandler.java

```diff
@@ -28,12 +29,18 @@
 public interface ExpressionHandler {
     /**
      * Create an {@link ExpressionEvaluator} that can evaluate the given <i>expression</i> on
-     * {@link io.delta.kernel.data.ColumnarBatch}s with the given <i>batchSchema</i>.
+     * {@link ColumnarBatch}s with the given <i>batchSchema</i>. The <i>expression</i> is
+     * expected to be a scalar expression where for each one input row there
+     * is a one output value.
      *
-     * @param batchSchema Schema of the input data.
+     * @param inputSchema Input data schema
      * @param expression  Expression to evaluate.
+     * @param outputType  Expected result data type.
      * @return An {@link ExpressionEvaluator} instance bound to the given expression and
-     * batchSchema.
+     * <i>inputSchem</i>.
      */
-    ExpressionEvaluator getEvaluator(StructType batchSchema, Expression expression);
+    ExpressionEvaluator getEvaluator(
+        StructType inputSchema,
+        Expression expression,
```

Comment:
Yeah, its likely we have to for aggregate expressions... because for aggregate .. would ExpressionsEvaluator produce ColumnVector?

Reply:
So it makes sense to make this a ScalaExpression

Reply:
Currently `Column` and `Literal` are not derived from `ScalarExpression`. Should we make them extend `ScalarExpression` before making this change? Otherwise we cannot populate partition values.

Reply:
This makes sense to me, seems like literal and column should be scalar expressions

Reply:
Okay good to merge. Let's follow up on it with @tdas next week?